***Выберете тест и проверьте, есть ли различия между выборками:***

**Задача 1.**

   Даны две независимые выборки. Не соблюдается условие нормальности
x1 380,420, 290
y1 140,360,200,900
Сделайте вывод по результатам, полученным с помощью функции, имеются ли статистические различия между группами?

Поскольку *две независимые выборки*, используем тест Манна-Уитни.

In [18]:
import scipy.stats as stats
import numpy as np 

x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])

stats.mannwhitneyu(x1, y1)


MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

p-value равно 0.6286, это больше уровня значимости 0.05. Значит нет статистически значимых различий между выборками. 

---

**Задача 2.**

Исследовалось влияние препарата на уровень давления пациентов. Сначала
измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть
ли статистически значимые различия между измерениями давления? В выборках не соблюдается условие нормальности.
1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150, 130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125

Используем анализ повторных измерений. Критерий Фридмана.

In [19]:
A = np.array([150, 160, 165, 145, 155])
B = np.array([140, 155, 150, 130, 135])
C = np.array([130, 130, 120, 130, 125])

stats.friedmanchisquare(A,B,C)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

p-value равно 0.0083, это меньше уровня значимости 0.05. Значит есть статистически значимые различий между выборками. 

---

**Задача 3.**


Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было. Есть
ли статистически значимые различия между измерениями давления?

Поскольку *две зависимые выборки*, используем тест Уилкоксона.

In [20]:
A = np.array([150, 160, 165, 145, 155])
B = np.array([140, 155, 150, 130, 135])

stats.wilcoxon(A,B)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

p-value равно 0.0625, это больше уровня значимости 0.05. Значит нет статистически значимых различий между выборками.

---

**Задача 4.**

Даны 3 группы учеников плавания. Не соблюдается условие нормальности.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54
Есть
ли статистически значимые различия между группами?

Поскольку *три независимые выборки*, используем тест Краскела-Уоллиса.

In [21]:
A = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
B = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
C = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(A,B,C)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

p-value равно 0.0650, это больше уровня значимости 0.05. Значит нет статистически значимых различий между группами.



---

**Задача 5.**

Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить
данную гипотезу, если известно, что размеры изделий подчинены нормальному закону
распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

Поскольку распределение нормальное и 𝞼  генеральной совокупности неизвестна, используем t - критерий (критерий Стьюдента) с учетом уровня значимости α/2 и числа степеней свободы k = n-1:

Гипотеза Н0 - среднее значение размеров изделий = 2,5 см

Гипотеза Н1 - среднее значение размеров изделий != 2,5 см

In [22]:
mu, alpha, n = 2.5, 0.05/2, 10
k = n - 1
size = [2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]
x_average = sum(size) / len(size)

temp = 0
for i in range(len(size)):
    temp += (size[i] - x_average)**2
sko = (temp / k)**(1/2)
t = round((x_average - mu) / (sko/ n**(1/2)),3)

print(f't-статистика = {t}')
print(f'Число степеней свободы "k" = {k}')

t-статистика = 0.563
Число степеней свободы "k" = 9


Критическое значение при α/2 = 0.025 и числа степеней свободы k = 9: t_крит = 2.262

Поскольку |t| < t_крит, принимается гипотеза Н0, т.е. ***среднее значение размеров изделий = 2,5 см*** 